src/root.zig

```diff
@@ -858,24 +858,33 @@ pub const Bookmark = opaque {
     }
 
     pub fn getTitle(self: *Bookmark, allocator: std.mem.Allocator) ![]u8 {
-        const len: usize = @intCast(FPDFBookmark_GetTitle(@ptrCast(self), null, 0));
-        const buffer = try allocator.alloc(u8, len);
-        const check = FPDFBookmark_GetTitle(@ptrCast(self), buffer.ptr, @intCast(len));
-
-        if (builtin.mode == .Debug) {
-            // Sanity checks that pdfium works as documented.
-            if (check != len) {
-                log.err("FPDFBookmark_GetTitle returned {d} but expected {d}", .{ check, len });
-                return error.Failed;
-            }
+        const size = FPDFBookmark_GetTitle(@ptrCast(self), null, 0);
+
+        if (size > 0) {
+            const u16_size: usize = @intCast(size / 2);
+            const u16_buffer = try allocator.alloc(u16, u16_size);
+            defer allocator.free(u16_buffer);
+
+            const check = FPDFBookmark_GetTitle(@ptrCast(self), @ptrCast(u16_buffer.ptr), size);
+
+            if (builtin.mode == .Debug) {
+                // Sanity checks that pdfium works as documented.
+                if (check != size) {
+                    log.err("FPDFBookmark_GetTitle returned {d} but expected {d}", .{ check, size });
+                    return error.Failed;
+                }
 
-            if (!std.mem.eql(u8, buffer[len - 2 ..], &UTF16_NUL)) {
-                log.warn("Unexpected behaviour: pdfium response was not terminated with UTF-16 NUL characters", .{});
-                return error.Failed;
+                if (u16_buffer.len > 0 and u16_buffer[u16_buffer.len - 1] != 0) {
+                    log.warn("Unexpected behaviour: pdfium response was not terminated with UTF-16 NUL character", .{});
+                    return error.Failed;
+                }
             }
-        }
 
-        return buffer;
+            const utf16_data = u16_buffer[0 .. u16_buffer.len - 1];
+            return try std.unicode.utf16LeToUtf8Alloc(allocator, utf16_data);
```

Comment:
If the user actually wanted utf16 then this would force them to do a redundant conversion back but I do appreciate the convenience.  

Maybe we do this:
```
fn getTitleUtf16() {
    ...
}

fn getTitleUtf8() {
    const utf16 = self.getTitleUtf16()
    // convert
}
```